### PR TITLE
Add some bots an deals endpoints

### PR DESCRIPTION
**Pull request by mschmuckler**
Adds 3 new endpoints. I tested that the param input and return values are correct and that these work by deploying them to a proxy server.

### CODE REVIEW

**Comment by mschmuckler** [comment]
@kirosc

File: src/index.ts [hunk 1]
```diff
@@ -268,6 +268,41 @@ export class API {
...
+      scope,
+      order,
+      order_direction: orderDirection,
+    };
+
+    if (botId) {
+      params['bot_id'] = botId;
+    }
+    return await this.request('GET', 1, '/deals', params);
+  }
```
Comment (kirosc) [inline src/index.ts:304]: Prefer allow user to pass in a `params` object and define the type in [here](https://github.com/kirosc/3commas-typescript/blob/master/src/types/types.ts)

So the function signature does not grow with the endpoint parameters

**Review by kirosc — CHANGES_REQUESTED** [review]
Thanks 🙏🏻 

Please also update the docs after change.

File: src/index.ts [hunk 2]
```diff
@@ -268,6 +268,41 @@ export class API {
     });
   }
 
+  async getBots() {
+    return await this.request('GET', 1, '/bots');
+  }
```
[the remark this answers is not in the document]
Comment (mschmuckler) [reply src/index.ts]: Updated with proper typings, and the defaults passed in match 3commas documentation

File: src/index.ts [hunk 3]
```diff
@@ -268,6 +268,41 @@ export class API {
...
+
+  async getBotsStats() {
+    return await this.request('GET', 1, '/bots/stats');
+  }
```
[the remark this answers is not in the document]
Comment (mschmuckler) [reply src/index.ts]: Updated with proper typings, and the defaults passed in match 3commas documentation

File: src/index.ts (hunk 1 above)
Comment (mschmuckler) [reply src/index.ts]: Updated with proper typings, and the defaults passed in match 3commas documentation

**Review by kirosc — APPROVED** [review]
🚀

**Pull request merged**